### PR TITLE
SSL option, .env file settings and custom llama path

### PR DESCRIPTION
**Pull request by cryptocake**
- Added SSL functionality, this can be enabled or disabled using the new .env file.
- Custom SSL certs can be added or use the one already provided in `cert/`.
- You have to set your llama.cpp installation directory in the .env file (required action).
- You can now choose which port gpt-llama.cpp will listen to.
- README.md now contains the additional installation steps regarding `.env`.

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# .env file
+################################################################################
+### GPT-LLAMA - GENERAL SETTINGS
+################################################################################
+
+## [REQUIRED] LLAMA_PATH - The absolute path of llama.cpp installation directory.
+LLAMA_PATH="/mnt/set/this/to/llama.cpp"
```
Comment (keldenl) [inline .env.template:7]: i actually avoided this so we can put this in the API KEY. less work for the end user to set up. i would prefer if we didn't move the llama_path to .env for now

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# .env file
+################################################################################
```
Comment (keldenl) [inline .env.template:2]: can we add some directions here that tells the user to copy this into their own .env file?

File: README.md [hunk 3]
```diff
@@ -140,13 +140,24 @@ Open another terminal window and test the installation by running the below scri
    npm install
    ```
 
-3. Start the server!
+3. Configure GPT-LLAMA:
```
Comment (keldenl) [inline README.md:143]: lets remove the llama_path instructions

File: cert/cert.crt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```
Comment (keldenl) [inline cert/cert.crt:1]: is pushing cert and private key the best thing we can do for security? doesn't seem like a good idea. what are our alternatives?

File: routes/chatRoutes.js [hunk 5]
```diff
@@ -76,7 +76,7 @@ const router = express.Router();
 
 router.post('/completions', async (req, res) => {
 	const modelId = req.body.model; // TODO: Implement model somehow
-	const llamaPath = getLlamaPath(req, res);
```
Comment (keldenl) [inline routes/chatRoutes.js:79]: same comment about llamapath

File: routes/embeddingsRoutes.js [hunk 6]
```diff
@@ -61,7 +61,7 @@ const router = express.Router();
 
 router.post('/', async (req, res) => {
 	const modelId = req.body.model; // TODO: Implement model somehow
-	const llamaPath = getLlamaPath(req, res);
+	const llamaPath = getLlamaPath();
```
Comment (keldenl) [inline routes/embeddingsRoutes.js:64]: same comment about llamapath

File: utils.js [hunk 7]
```diff
@@ -96,10 +105,8 @@ export const getModelName = (path) => {
 	return modelArr[modelArr.length - 1];
 };
 
-export const getLlamaPath = (req, res) => {
-	const modelPath = getModelPath(req, res);
-	const path = modelPath.split('llama.cpp')[0]; // only
-	return join(path, 'llama.cpp');
+export const getLlamaPath = () => {
```
Comment (keldenl) [inline utils.js:108]: same comment about llamapath

File: utils.js [hunk 8]
```diff
@@ -1,6 +1,15 @@
 import { sep, join, resolve } from 'path';
 import { nanoid } from 'nanoid';
 import { readdir } from 'fs/promises';
+import fs from 'fs';
+
+if (!fs.existsSync('.env')) {
+  console.error('.env file not found. Please create a .env file.');
+  process.exit(1);
+}
```
Comment (keldenl) [inline utils.js:9]: do we ever need this? if this is false it should fail on index.js already right?

**Review by keldenl — CHANGES_REQUESTED** [review]
left some comments

**Comment by cryptocake** [comment]
The changes in this PR are obsolete now. Closing.

**Pull request closed without merging**